### ISSUE TITLE
application/json default charset is not UTF-8

### ISSUE THREAD

**Issue by vonox7** [Kotlin/ktor#80]
[RFC4627 Section 3](https://tools.ietf.org/html/rfc4627#section-3) specifies that the default encoding of `application/json` is UTF-8. Ktor falls back to ISO-8859-1 (see [RequestContent.kt:11](https://github.com/Kotlin/ktor/blob/master/ktor-core/src/org/jetbrains/ktor/request/RequestContent.kt#L11)).

Further, the [application/json entry on IANA](https://www.iana.org/assignments/media-types/application/json) has the note:
```
Note:  No "charset" parameter is defined for this registration.
Adding one really has no effect on compliant recipients.
```

Ktor doesn't comply here to the standard and is to be considered as non-compliant server. See discussions on this issue e.g. on https://github.com/request/request/issues/383.

Please fallback on `application/json` to UTF-8.

**Comment by cy6erGn0m** [comment]
getContentAsString just handles content as it is a regular text content, not sure if we have to apply particular content specific rules here otherwise it will become too complex and unpredictable. However we definitely have missed this thing in few places so I believe we have to fix it and think about this use-case while we are going to provide request body transformation (as we already do for response content).

**Comment by orangy** [comment]
This has been addressed in 0.4.0, when all `respondText` and `respondWrite` calls results in UTF-8 by default, unless a different encoding was specified explicitly.
